### PR TITLE
Fix EZP-24556: Solr search engine: CustomField criterion is not fully handled

### PR DESCRIPTION
**Pull request by pspanja**
> Review dependency graph
> * https://github.com/ezsystems/ezpublish-kernel/pull/1305 Main languages core
>  * https://github.com/ezsystems/ezpublish-kernel/pull/1322 Multicore tests
>    * **https://github.com/ezsystems/ezpublish-kernel/pull/1326 Complete CustomField support**
>       * https://github.com/ezsystems/ezpublish-kernel/pull/1327 Extract CoreFilter service
>       * https://github.com/ezsystems/ezpublish-kernel/pull/1333 Indexable Selection
>          * https://github.com/ezsystems/ezpublish-kernel/pull/1334 Indexable RelationList
> * https://github.com/ezsystems/ezpublish-kernel/pull/1329 DateModified sort clause

#### Based on https://github.com/ezsystems/ezpublish-kernel/pull/1322

`CustomField` criterion was not fully supported in Solr search engine, missing visiting for range operators and Location Search. This implements the missing parts with integration tests.

#### Additionally fixed

1. Elasticsearch: `CustomField` range visiting was using filter range, which did not analyze the query terms. For keyword/lowercase analyizer, this was resulting in case-sensitive range search.
2. Solr: added missing parenthesis to `ContentTypeId` criterion visiting in Location Search
3. Solr: removed unnecessary parenthesis in `LogicalNot` criterion visiting
4. Added case-insensitivity to field type search tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -53,6 +53,7 @@ branches:
   only:
     - master
     - impl-EZP-24434-aa-core-2
+    - fix-EZP-24556-solr-customfield
```
Comment (andrerom) [inline .travis.yml]: :)

**Comment by andrerom** [comment]
+1

File: .travis.yml (hunk 1 above)
Comment (pspanja) [inline .travis.yml]: Removed in https://github.com/ezsystems/ezpublish-kernel/commit/bbc43859858b474071cf403edb762a58e348550b

**Comment by pspanja** [comment]
Manually merged in https://github.com/ezsystems/ezpublish-kernel/commit/bbc43859858b474071cf403edb762a58e348550b

**Pull request closed without merging**